### ISSUE TITLE
Global Translation JSON with Inline Loader in NX Lib?

### ISSUE THREAD

**Issue by ChazUK** [ngneat/transloco#325]
I'm using Transloco within an NX Mono Repo, which has a shared library "UI". I've set up Transloco so that each component uses a scoped inline loader to ensure that it's easy to track translations between the components. However there are a number of translations which match across components so it would be good if they could be extracted out and contained in a centralised file to keep it DRY.

When I add a scope to the UiModule the translation files aren't loaded, and can't be used in other inline loaded files.

I've got a demo setup here https://github.com/ChazUK/nx-transloco-global-translations Just need to run `ng serve demo-app`

global - en.json
```
{
  "title": "Welcome to Demo App!",
  "button": "Button",
  "hello": "Hola"
}
```

button - en.json
```
{
  "hello-friend": "{{ ui.hello }} Friend",
  "text": "English Button"
}
```

button.component.ts
```
@Component({
  selector: 'nx-transloco-global-translations-button',
  template: `
    <ng-template transloco let-t>
      <h4>{{ t('button.hello-friend') }}</h4>
      <button>{{ t('ui.button') }} Text: {{ t('button.text') }}</button>
    </ng-template>
  `,
  providers: [
    {
      provide: TRANSLOCO_SCOPE,
      useValue: {
        scope: 'button',
        loader: loader((lang, root) => import(`./${root}/${lang}.json`)),
      },
    },
  ],
})
export class ButtonComponent {}
```

**Comment by jurienhamaker** [comment]
Experiencing the same issue. When I use `multi: true` for this setup I get `Cannot mix multi providers and regular providers`

**Comment by ArielGueta** [comment]
![image](https://user-images.githubusercontent.com/20522870/89437317-16513e00-d750-11ea-849c-f50b00376582.png)


Try to move the scope provider to the button component.

**Comment by ChazUK** [comment]
@ArielGueta The Button Component already has a Scope and Loader.

**Comment by ArielGueta** [comment]
So what's the scope in the UIModule?

**Comment by ChazUK** [comment]
I want to scope the UiModule because it's a library, so scoping it would hopefully mean it wouldn't clash with anything. Which I'm pretty sure is quite standard

**Comment by ArielGueta** [comment]
You can't. It's not related to Transloco, it's how Angular works.

**Comment by ChazUK** [comment]
@ArielGueta: As far as I'm aware adding a scope to a component will wrap the loaded translations in the provided key, so in my example I have the button component scoped to "button". So JSON translation file `{ "text": "Button Text" }` will be loaded into Transloco as `{ "button": { "text": "Button Text" } }`. And in the template I can either define the scope, or use the full path to load the desired string.

I feel that the "Global" translation file I want to create, which can be shared among all components within the module, should be able to have a scope so that the keys don't clash and overwrite each other. That in itself is directly related to Transloco.

The screenshot you posted above ([link for future reference](https://netbasal.gitbook.io/transloco/lazy-load-translation-files/scope-configuration)) states that scoping follows the Angular Dependancy Injection rules, but doesn't necessarily mean the  functionality I described can't be achieved.

@jurienhamaker: I've managed to find a quick workaround that still inline loads translations.

```
@Injectable({ providedIn: 'root' })
export class CustomLoader implements TranslocoLoader {
  getTranslation(lang: string) {
    return import(`./i18n/${lang}.json`);
  }
}

@NgModule({
  imports: [
    TranslocoModule,
    ...
  ]
  providers: [
    PageContentResolver,
    {
      provide: TRANSLOCO_LOADER,
      useClass: CustomLoader,
    },
    ...
  ]
  ...
})
export class UiModule {}
```
